vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/authorization/restrictusers/subjectchecker.go

```diff
@@ -104,6 +106,13 @@ func (ctx *RoleBindingRestrictionContext) groupsForUser(subject rbac.Subject) ([
 		return []*userv1.Group{}, fmt.Errorf("not a user: %q", subject.Name)
 	}
 
+	err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
```

Comment:
I would rather expect this much higher up in the outer plumbing code that sets up the group cache, and therefore delay the readiness of the whole apiserver.

Reply:
> I would rather expect this much higher up in the outer plumbing code that sets up the group cache, and therefore delay the readiness of the whole apiserver.

that's an undesireable outcome.  This affects a minor part of a minor feature that is ultimately optional.  Delaying readiness of the server is inappropriate in this case.  The authentication side fails closed and isn't always used either, so it's also inappropriate to delay readiness.

Also, if you delayed readiness, you would create a bootstrap loop with the openshift apiserver that would never be resolved on installation.

Reply:
Yes, we realized the bootstrap loop in slack discussion. Some delay here is probably the best we can do if we care about new kube-apiservers to give correct results of the admission plugin. Any other suggestion?